lib/models/order.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+
+const Promise = require('bluebird')
+const Model = require('../model')
+let lastCID = Date.now()
+
+class Order extends Model {
+  constructor (data, transport) {
+    super(data)
+
+    this._onTransportUpdate = this._onTransportUpdate.bind(this)
+    this._onTransportClose = this._onTransportClose.bind(this)
+
+    if (transport) {
+      transport.onOrderUpdate({
+        symbol: this.symbol,
+        cbGID: this.gid || null
+      }, this._onTransportUpdate)
+
+      transport.onOrderClose({
+        symbol: this.symbol,
+        cbGID: this.gid || null
+      }, this._onTransportClose)
+    }
+
+    this._transport = transport
+  }
+
+  submit () {
+    if (!this._transport) return Promise.reject('no transport')
+
+    return this._transport.submitOrder(this)
+  }
+
+  _onTransportUpdate (order) {
+    if (order[2] !== this.cid) return
+
+    Object.assign(this, Order.unserialize(order))
+  }
+
+  _onTransportClose (order) {
+    if (order[2] !== this.cid) return
+
+    Object.assign(this, Order.unserialize(order))
+  }
+
+  serialize () {
+    return [
+      this.id,
+      this.gid,
+      this.cid,
+      this.symbol,
+      this.mtsCreate,
+      this.mtsUpdate,
+      this.amount,
+      this.amountOrig,
+      this.type,
+      this.typePrev,
+      null,
+      null,
+      this.flags,
+      this.status,
+      null,
+      null,
+      this.price,
+      this.priceAvg,
+      this.priceTrailing,
+      this.priceAuxLimit,
+      null,
+      null,
+      null,
+      this.notify ? 1 : 0,
+      this.hidden ? 1 : 0,
+      this.placedId
+    ]
+  }
+
+  toNewOrderPacket () {
+    return {
+      gid: this.gid,
+      cid: this.cid || lastCID++,
```

Comment:
lastCID is undefined

Reply:
It's defined at the top of the file & not exposed; I did this since using `Date.now()` for the `cid` of two orders in rapid succession yields the same `cid`. This way at least the default is unique per run.

Reply:
ah makes sense!